netmiko/yamaha/yamaha.py

```diff
@@ -0,0 +1,101 @@
+from netmiko.base_connection import BaseConnection
+from netmiko.ssh_exception import NetmikoTimeoutException
+from netmiko import log
+import time
+
+
+class YamahaBase(BaseConnection):
+    def session_preparation(self):
+        """Prepare the session after the connection has been established."""
+        self._test_channel_read(pattern=r"[>#]")
+        self.set_base_prompt()
+        self.disable_paging(command="console lines infinity")
+        time.sleep(0.3 * self.global_delay_factor)
+        self.clear_buffer()
+
+    def check_enable_mode(self, *args, **kwargs):
+        """No enable mode on Yamaha."""
+        pass
+
+    def enable(self, *args, **kwargs):
+        """No enable mode on Yamaha."""
+        pass
+
+    def exit_enable_mode(self, *args, **kwargs):
+        """No enable mode on Yamaha."""
+        pass
+
+    def check_config_mode(self, check_string="#", pattern=""):
+        """Checks if the device is in administrator mode or not."""
+        return super(YamahaBase, self).check_config_mode(
+            check_string=check_string, pattern=pattern
+        )
+
+    def config_mode(self, config_command="administrator", pattern="ssword"):
+        """Enter into administrator mode and configure device."""
+        output = ""
+        msg = (
+            "Failed to enter administrator mode. Please ensure you pass "
+            "the 'secret' argument to ConnectHandler."
+        )
+        if not self.check_config_mode():
+            self.write_channel(self.normalize_cmd(config_command))
+            try:
+                output += self.read_until_prompt_or_pattern(pattern=pattern)
+                self.write_channel(self.normalize_cmd(self.secret))
```

Comment:
I think this probably should just be moved into the `enable()` call. So you have some sort of an `administrator` mode that requires you pass a second password and you need those privileges to make configuration changes. I think that should be moved to the `enable()` call (i.e. that probably is more consistent to Netmiko behavior than having this embedded in the `config_mode` call).

I am also thinking if we do this, then we can eliminate a bunch of code and rely more on `super()` calls to the base class.